src/cargo/core/compiler/build_plan.rs

```diff
@@ -0,0 +1,143 @@
+//! A graph-like structure used to represent the rustc commands to build the project and the
+//! interdependencies between them.
+//!
+//! The BuildPlan structure is used to store the dependency graph of a dry run so that it can be
+//! shared with an external build system. Each Module in the BuildPlan comprises a single
+//! subprocess and defines the build environment, the outputs produced by the subprocess, and the
+//! dependencies on other Modules.
+
+use std::collections::BTreeMap;
+
+use super::{Context, Unit};
+use super::context::OutputFile;
+use util::{internal, CargoResult, ProcessBuilder};
+use std::sync::Arc;
+use std::path::PathBuf;
+use serde_json;
+
+#[derive(Debug, Default, Serialize)]
+struct Module {
```

Comment:
Would a better name for this perhaps be something like `Compilation` or `Invocation`? A "module" in Rust is pretty different from what this is I believe